x-pack/plugins/alerting/server/task_runner/task_runner.ts

```diff
@@ -708,6 +712,30 @@ export class TaskRunner<
     ruleMonitoring.execution.calculated_metrics.success_ratio =
       ruleMonitoring.execution.history.filter(({ success }) => success).length /
       ruleMonitoring.execution.history.length;
+
+    // Percentile calculations for rule executions, limited to predefined number of samples
+    const historyWithDuration = ruleMonitoring.execution.history.filter(
+      (history) => typeof history.duration === 'number'
+    );
+    // Reformat history array to numbers array (durations in ms)
+    const durationSamples = historyWithDuration.map((history) => history.duration);
```

Comment:
I think, we can avoid to iteration on the history[]
```
ruleMonitoring.execution.history.reduce((duration, hist) => {
  if (typeof hist.duration === 'number') {
    return [...duration, hist.duration];
  }
 return duration;
}, []);
```
if you do not like the spreading, we can just do a for loop with a `const duration = []` and add to it. That might be better.
 what do you think?